cvat-core/src/server-proxy.ts

```diff
@@ -165,13 +165,48 @@ async function chunkUpload(file: File, uploadConfig): Promise<{ uploadSentSize:
     });
 }
 
-function generateError(errorData: AxiosError<{ message?: string }>): ServerError {
+function generateError(errorData: AxiosError): ServerError {
     if (errorData.response) {
-        if (errorData.response.data?.message) {
-            return new ServerError(errorData.response.data?.message, errorData.response.status);
+        if (errorData.response.status >= 500 && typeof errorData.response.data === 'string') {
+            return new ServerError(errorData.response.data, errorData.response.status);
         }
-        const message = `${errorData.message}. ${JSON.stringify(errorData.response.data || '')}.`;
-        return new ServerError(message, errorData.response.status);
+
+        if (errorData.response.status >= 400 && errorData.response.data) {
+            // serializer.ValidationError
+            if (typeof errorData.response.data === 'object') {
+                const generalFields = ['non_field_errors', 'detail', 'message'];
+                const generalFieldsHelpers = {
+                    'Invalid token.': 'Not authenticated request, try to login again',
+                };
+
+                for (const field of generalFields) {
+                    if (field in errorData.response.data) {
+                        const message = errorData.response.data[field].toString();
+                        return new ServerError(
+                            generalFieldsHelpers[message] || message,
+                            errorData.response.status,
+                        );
+                    }
+                }
+
+                // serializers fields
+                const message = Object.keys(errorData.response.data).map((key) => (
+                    `**${key}**: ${errorData.response.data[key].toString()}`
+                )).join('\n\n');
```

Comment:
Why double `\n`?

Reply:
That is markdown. Single \n does not set new line